Sources/Kodable/Codable+Lossless.swift

```diff
@@ -38,14 +38,35 @@ struct LosslessValue<T: LosslessDecodable>: Decodable {
             decode(Float.self),
         ]
 
-        guard let rawValue = types.lazy.compactMap({ $0(decoder) }).first, let value = T("\(rawValue)") else {
+        guard let anyLosslessDecodable = T.self as? LosslessDecodable.Type, let rawValue = types.lazy.compactMap({ $0(decoder) }).first, let value = anyLosslessDecodable.init("\(rawValue)") as? T else {
             throw Corrupted()
         }
-
+        
         self.value = value
     }
 }
 
+// MARK: Helper type to decode lossy arrays
+
+struct LossyDecodableArray<Element: Decodable>: Decodable {
+    private struct ElementWrapper: Decodable {
+        var element: Element?
+
+        init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            self.element = try? container.decode(Element.self)
+        }
+    }
+
+    var elements: [Element]
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let wrappers = try container.decode([ElementWrapper].self)
+        self.elements = wrappers.compactMap(\.element)
+    }
```

Comment:
I'd add 1 extra test case, which would be the exact case that we saw a failure on:

Try to parse an array of structs where 1 or more elements of the array fail to parse entirely. So we test this exact code snippet over here

Reply:
Edit: nevermind, now I see that `nam` would fail the `name` parsing. I overlooked that haha such similar naming. Perhaps name it after `dragonite` or something, so it's more readable that it would obviously fail to be parsed. Using more descriptive values would also help, e.g. `"dragonite": "this key will fail to be parsed"` 👍 I try to follow those rules when writing tests